src/overmap_ui.cpp

```diff
@@ -1429,12 +1429,21 @@ static tripoint display( const tripoint &orig, const draw_data_t &data = draw_da
                 curs.y = p.y;
             }
         } else if( action == "CHOOSE_DESTINATION" ) {
+            bool in_road_vehicle = g->u.in_vehicle && g->u.controlling_vehicle;
+            const optional_vpart_position vp = g->m.veh_at( g->u.pos() );
+            bool in_boat = false;
+            if( vp && in_road_vehicle ) {
+                vehicle *const veh = &vp->vehicle();
```

Comment:
```suggestion
                vehicle &veh = vp->vehicle();
```
And adjust the line below accordingly.

----

I'll never get why people want to use pointers all the time, when they already have a reference.

Reply:
I think the phrase is "cargo-cult programming"? I search for other places in the code where it checks for the presence of a vehicle, and I copy it because I assume theres a good reason why it is that way. Guess there isnt. I'll change it.